# ANALYSIS OF LENGTH OF STAY AT HOSPITALS OF CORONA VIRUS PATIENTS

### 1.What factors affect the length of stay at the hospitals?
### 2.Does Increasing age lead to inrease the stay at the hospitals?
### 3.Does the amount one deposits give access to certain types of benefits?
### 4.Whats the relationship between severity of illness and type of admission? What age groups normally fall here
### 5.Which hospitals in which regions have fewer beds?
### 6.Which hospitals or regions have the highest admission deposit?

In [73]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [74]:
data=pd.read_csv('train.csv')
data.head()

case_id  Hospital_code Hospital_type_code  City_Code_Hospital  \
0        1              8                  c                   3   
1        2              2                  c                   5   
2        3             10                  e                   1   
3        4             26                  b                   2   
4        5             26                  b                   2   

  Hospital_region_code  Available Extra Rooms in Hospital    Department  \
0                    Z                                  3  radiotherapy   
1                    Z                                  2  radiotherapy   
2                    X                                  2    anesthesia   
3                    Y                                  2  radiotherapy   
4                    Y                                  2  radiotherapy   

  Ward_Type Ward_Facility_Code  Bed Grade  patientid  City_Code_Patient  \
0         R                  F        2.0      31397                7.0   
1         S                  F        2.0      31397                7.0   
2         S                  E        2.0      31397                7.0   
3         R                  D        2.0      31397                7.0   
4         S                  D        2.0      31397                7.0   

  Type of Admission Severity of Illness  Visitors with Patient    Age  \
0         Emergency             Extreme                      2  51-60   
1            Trauma             Extreme                      2  51-60   
2            Trauma             Extreme                      2  51-60   
3            Trauma             Extreme                      2  51-60   
4            Trauma             Extreme                      2  51-60   

   Admission_Deposit   Stay  
0             4911.0   0-10  
1             5954.0  41-50  
2             4745.0  31-40  
3             7272.0  41-50  
4             5558.0  41-50

In [75]:
data.shape

(318438, 18)

In [76]:
#Check for missing values
data.isnull().sum()

case_id                                 0
Hospital_code                           0
Hospital_type_code                      0
City_Code_Hospital                      0
Hospital_region_code                    0
Available Extra Rooms in Hospital       0
Department                              0
Ward_Type                               0
Ward_Facility_Code                      0
Bed Grade                             113
patientid                               0
City_Code_Patient                    4532
Type of Admission                       0
Severity of Illness                     0
Visitors with Patient                   0
Age                                     0
Admission_Deposit                       0
Stay                                    0
dtype: int64

#### Missing values have to be dealt with first

In [77]:
#only 113 missing values, so they can de dropped without affecting the analysis
data=data.dropna(subset=['Bed Grade'])

In [78]:
data.isnull().sum()

case_id                                 0
Hospital_code                           0
Hospital_type_code                      0
City_Code_Hospital                      0
Hospital_region_code                    0
Available Extra Rooms in Hospital       0
Department                              0
Ward_Type                               0
Ward_Facility_Code                      0
Bed Grade                               0
patientid                               0
City_Code_Patient                    4532
Type of Admission                       0
Severity of Illness                     0
Visitors with Patient                   0
Age                                     0
Admission_Deposit                       0
Stay                                    0
dtype: int64

In [79]:
data['City_Code_Patient'].value_counts()

8.0     123977
2.0      38835
1.0      26357
7.0      23797
5.0      20079
4.0      15379
9.0      11794
15.0      8943
10.0      8172
6.0       6005
12.0      5646
3.0       3772
23.0      3697
14.0      2927
16.0      2252
13.0      1625
21.0      1602
20.0      1409
18.0      1404
19.0      1028
26.0      1023
25.0       798
27.0       771
11.0       658
28.0       521
22.0       405
24.0       360
30.0       133
29.0        98
33.0        78
31.0        59
37.0        57
32.0        52
34.0        46
35.0        16
36.0        12
38.0         6
Name: City_Code_Patient, dtype: int64

In [80]:
#The mode seems to be the best value to use to fill the NaN values
city_mode=data['City_Code_Patient'].mode()[0]

In [81]:
data['City_Code_Patient']=data['City_Code_Patient'].fillna(city_mode)

In [82]:
data.isnull().sum()

case_id                              0
Hospital_code                        0
Hospital_type_code                   0
City_Code_Hospital                   0
Hospital_region_code                 0
Available Extra Rooms in Hospital    0
Department                           0
Ward_Type                            0
Ward_Facility_Code                   0
Bed Grade                            0
patientid                            0
City_Code_Patient                    0
Type of Admission                    0
Severity of Illness                  0
Visitors with Patient                0
Age                                  0
Admission_Deposit                    0
Stay                                 0
dtype: int64

#### Missing values have been taken care of

In [83]:
data.head()

case_id  Hospital_code Hospital_type_code  City_Code_Hospital  \
0        1              8                  c                   3   
1        2              2                  c                   5   
2        3             10                  e                   1   
3        4             26                  b                   2   
4        5             26                  b                   2   

  Hospital_region_code  Available Extra Rooms in Hospital    Department  \
0                    Z                                  3  radiotherapy   
1                    Z                                  2  radiotherapy   
2                    X                                  2    anesthesia   
3                    Y                                  2  radiotherapy   
4                    Y                                  2  radiotherapy   

  Ward_Type Ward_Facility_Code  Bed Grade  patientid  City_Code_Patient  \
0         R                  F        2.0      31397                7.0   
1         S                  F        2.0      31397                7.0   
2         S                  E        2.0      31397                7.0   
3         R                  D        2.0      31397                7.0   
4         S                  D        2.0      31397                7.0   

  Type of Admission Severity of Illness  Visitors with Patient    Age  \
0         Emergency             Extreme                      2  51-60   
1            Trauma             Extreme                      2  51-60   
2            Trauma             Extreme                      2  51-60   
3            Trauma             Extreme                      2  51-60   
4            Trauma             Extreme                      2  51-60   

   Admission_Deposit   Stay  
0             4911.0   0-10  
1             5954.0  41-50  
2             4745.0  31-40  
3             7272.0  41-50  
4             5558.0  41-50

In [84]:
data.dtypes

case_id                                int64
Hospital_code                          int64
Hospital_type_code                    object
City_Code_Hospital                     int64
Hospital_region_code                  object
Available Extra Rooms in Hospital      int64
Department                            object
Ward_Type                             object
Ward_Facility_Code                    object
Bed Grade                            float64
patientid                              int64
City_Code_Patient                    float64
Type of Admission                     object
Severity of Illness                   object
Visitors with Patient                  int64
Age                                   object
Admission_Deposit                    float64
Stay                                  object
dtype: object

#### Looks like some of the columns have the wrong datatypes and needs to be corrected

In [85]:
data['Hospital_code']=data['Hospital_code'].astype('object')

In [86]:
data['Available Extra Rooms in Hospital']=data['Available Extra Rooms in Hospital'].astype('object')

In [87]:
data['Bed Grade']=data['Bed Grade'].astype('object')

In [88]:
data['City_Code_Patient']=data['City_Code_Patient'].astype('object')

In [89]:
data['Visitors with Patient']=data['Visitors with Patient'].astype('object')

In [90]:
data.dtypes

case_id                                int64
Hospital_code                         object
Hospital_type_code                    object
City_Code_Hospital                     int64
Hospital_region_code                  object
Available Extra Rooms in Hospital     object
Department                            object
Ward_Type                             object
Ward_Facility_Code                    object
Bed Grade                             object
patientid                              int64
City_Code_Patient                     object
Type of Admission                     object
Severity of Illness                   object
Visitors with Patient                 object
Age                                   object
Admission_Deposit                    float64
Stay                                  object
dtype: object

#### The data types have all been taken care of 

### There seem to be a lot of duplicates in it, most of the patient id which is supposed to be unique are occuring like 30 to 50 times here

In [91]:
data['patientid'].value_counts()

66714     50
91292     43
38525     39
101359    36
33491     34
          ..
97969      1
98821      1
104966     1
129546     1
58524      1
Name: patientid, Length: 92009, dtype: int64

#### It can clear be seen that some patients were captured so many times in this, patient ID 66714 was captured 50 times

In [92]:
data.shape

(318325, 18)

In [93]:
data=data.drop_duplicates(subset=['patientid'])

In [94]:
sum(data['patientid'].duplicated())

0

In [95]:
data.shape

(92009, 18)

In [96]:
data['patientid'].value_counts()

2047      1
82493     1
111139    1
100900    1
98853     1
         ..
58678     1
46396     1
48445     1
42302     1
131072    1
Name: patientid, Length: 92009, dtype: int64

#### Good, now they are occuring just once, which is what we want

## The data cleaning has all been done

In [97]:
data=data.drop(columns=['Hospital_code','case_id','City_Code_Patient'])

In [99]:
data.to_csv('Cleaned_data.csv')